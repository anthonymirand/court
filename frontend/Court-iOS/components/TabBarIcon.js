import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'expo';

import Colors from '../constants/Colors';

/**
* Tab bar icon represnets a singular entry in the tab bar navigator
*/
export default class TabBarIcon extends React.Component {
  render() {
    return (
      <Icon.Ionicons
        name={this.props.name}
        size={26}
        style={{ marginBottom: -3 }}
        color={this.props.focused ? Colors.tabIconSelected : Colors.tabIconDefault}
      />
    );
  }
}

TabBarIcon.propTypes = {
  /**
  * Name of icon to render
  */
  name: PropTypes.string.isRequired,
  /**
  * Boolean indictating if tab icon is selected
  */
  focused: PropTypes.bool.isRequired,
}