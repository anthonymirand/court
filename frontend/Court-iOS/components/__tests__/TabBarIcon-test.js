import 'react-native';
import React from 'react';
import TabBarIcon from '../TabBarIcon';
import renderer from 'react-test-renderer';

it('Renders Default TabBarIcon', () => {
  const onPress = jest.fn();
  const tree = renderer.create(
    <TabBarIcon
      name="md-chatbubbles"
      focused={true}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
