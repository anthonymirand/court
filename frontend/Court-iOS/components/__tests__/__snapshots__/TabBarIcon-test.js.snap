// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default TabBarIcon 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
